lib/tree.rb

```diff
@@ -16,51 +16,114 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(log n)
+  # Space Complexity: O(1)
   def add(key, value)
```

Comment:
👍 The time complexity is O(log n) if the tree is balanced and O(n) if it's unbalanced.

The space complexity is the same (because of the recursion).